frontend/packages/metal3-plugin/src/components/baremetal-hosts/BareMetalHostDetails.tsx

```diff
@@ -73,8 +73,10 @@ const BareMetalHostDetails: React.FC<BareMetalHostDetailsProps> = ({
   const nodeName = getMachineNodeName(machine);
   const node = getMachineNode(machine, nodes);
   const role = getMachineRole(machine);
-  const RAMGB = humanizeDecimalBytes(getHostRAMMiB(host) * 2 ** 20).string;
-  const totalStorageCapacity = humanizeDecimalBytes(getHostTotalStorageCapacity(host)).string;
+  const hostRAM = getHostRAMMiB(host);
+  const RAMGB = hostRAM ? humanizeDecimalBytes(hostRAM * 2 ** 20).string : '-';
```

Comment:
There is a `DASH` constant in @console/shared

```suggestion
  const RAMGB = hostRAM ? humanizeDecimalBytes(hostRAM * 2 ** 20).string : DASH;
```

same for the other dashes below